ff/src/fields/models/fp/mod.rs

```diff
@@ -0,0 +1,1015 @@
+use ark_serialize::{
+    buffer_byte_size, CanonicalDeserialize, CanonicalDeserializeWithFlags, CanonicalSerialize,
+    CanonicalSerializeWithFlags, EmptyFlags, Flags, SerializationError,
+};
+use ark_std::{
+    cmp::{Ord, Ordering, PartialOrd},
+    fmt::{Display, Formatter, Result as FmtResult},
+    io::{Read, Result as IoResult, Write},
+    marker::PhantomData,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+    str::FromStr,
+    One, Zero,
+};
+
+#[macro_use]
+mod montgomery_backend;
+pub use montgomery_backend::*;
+
+use crate::{
+    BigInt, BigInteger, FftField, Field, FromBytes, LegendreSymbol, PrimeField, SquareRootField,
+    ToBytes,
+};
+/// A trait that specifies the configuration of a prime field.
+/// Also specifies how to perform arithmetic on field elements.
+pub trait FpConfig<const N: usize>: crate::FftConfig<Field = Fp<Self, N>> {
+    /// The modulus of the field.
+    const MODULUS: crate::BigInt<N>;
+
+    /// A multiplicative generator of the field.
+    /// `Self::GENERATOR` is an element having multiplicative order
+    /// `Self::MODULUS - 1`.
+    const GENERATOR: Fp<Self, N>;
+
+    /// Additive identity of the field, i.e. the element `e`
+    /// such that, for all elements `f` of the field, `e + f = f`.
+    const ZERO: Fp<Self, N>;
+
+    /// Multiplicative identity of the field, i.e. the element `e`
+    /// such that, for all elements `f` of the field, `e * f = f`.
+    const ONE: Fp<Self, N>;
```

Comment:
The replacement for `FpParameters`, `FpConfig`, is much simplified; only the bare essentials need to be specified here, compared to implementation-details like R, CAPACITY, etc earlier.